File: weblog-backend/weblog-module-content/src/main/java/com/blog/content/entity/Post.java
package com.blog.content.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.blog.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

/**
 * 文章实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_post")
public class Post extends BaseEntity {

    private String title;

    private String slug;

    private String summary;

    private String coverImage;

    private Long categoryId;

    private Long subCategoryId;

    private Long authorId;

    /** draft / published / scheduled */
    private String status;

    /** immediate / scheduled */
    private String publishType;

    private LocalDateTime scheduledTime;

    private Boolean isPublished;

    private Integer viewCount;

    private Integer likeCount;

    private Integer collectCount;

    private Integer commentCount;

    private String seoTitle;

    private String seoDescription;

    private String seoKeywords;

    /** 预览主题 */
    private String previewTheme;

    /** 代码主题 */
    private String codeTheme;

    private Boolean isTop;

    /** 是否禁用（禁用后前台不展示） */
    private Boolean isDisabled;
}

File: weblog-user/api/content/post.ts
import { http } from '~/utils/http'

export interface PostVO {
  id: number
  title: string
  slug: string
  summary: string | null
  coverImage: string | null
  categoryId: number | null
  categoryName: string | null
  subCategoryId: number | null
  subCategoryName: string | null
  authorId: number
  authorNickname: string | null
  authorAvatar: string | null
  status: string
  viewCount: number
  likeCount: number
  collectCount: number
  commentCount: number
  isTop: boolean
  createTime: string
  updateTime: string
  content?: string
  htmlContent?: string
  tags: TagVO[]
  seoTitle?: string
  seoDescription?: string
  seoKeywords?: string
  previewTheme?: string
  codeTheme?: string
}

export interface TagVO {
  id: number
  name: string
  slug: string
  color: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export interface PostDetailResult {
  post: PostVO
  prev: { id: number; title: string; slug: string; coverImage?: string } | null
  next: { id: number; title: string; slug: string; coverImage?: string } | null
}

export const postApi = {
  /** 文章列表 */
  list: (params: { pageNum?: number; pageSize?: number; categoryId?: number; tagId?: number; categorySlug?: string; tagSlug?: string; sortBy?: string }) =>
    http.get<any, { data: PageResult<PostVO> }>('/portal/post', { params }),

  /** 文章详情（含上下篇） */
  detail: (slug: string) =>
    http.get<any, { data: PostDetailResult }>(`/portal/post/${slug}`),

  /** 今日发布文章列表 */
  listToday: (limit = 8) =>
    http.get<any, { data: PostVO[] }>('/portal/post/today', { params: { limit } }),

  /** 最近发布文章列表 */
  listRecent: (limit = 10) =>
    http.get<any, { data: PostVO[] }>('/portal/post/recent', { params: { limit } }),

  /** 相关文章推荐 */
  getSimilarPosts: (postId: number) =>
    http.get<any, { data: PostVO[] }>(`/portal/post/${postId}/similar`),
}
